aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/domain/options/ListInstancesOptions.java

```diff
@@ -0,0 +1,258 @@
+   }
+
+   public ListInstancesOptions vSwitchId(String vSwitchId) {
+      queryParameters.put(VSWITCH_ID_PARAM, vSwitchId);
+      return this;
+   }
+
+   public ListInstancesOptions zoneId(String zoneId) {
+      queryParameters.put(ZONE_ID_PARAM, zoneId);
+      return this;
+   }
+
+   public ListInstancesOptions instanceIds(String... instanceIds) {
+      String instanceIdsAsString = Joiner.on(",")
+              .join(Iterables.transform(Arrays.asList(instanceIds), new Function<String, String>() {
+                 @Override
+                 public String apply(String s) {
+                    return new StringBuilder(s.length() + 1).append('"').append(s).append('"').toString();
+                 }
+              }));
+      queryParameters.put(INSTANCE_IDS_PARAM, String.format("[%s]", instanceIdsAsString));
+      return this;
+   }
+
+   public ListInstancesOptions instanceType(String instanceType) {
+      queryParameters.put(INSTANCE_TYPE_PARAM, instanceType);
+      return this;
+   }
+
+   public ListInstancesOptions instanceTypeFamily(String instanceTypeFamily) {
+      queryParameters.put(INSTANCE_TYPE_FAMILY_PARAM, instanceTypeFamily);
+      return this;
+   }
+
+   public ListInstancesOptions instanceNetworkType(String instanceNetworkType) {
+      queryParameters.put(INSTANCE_NETWORK_TYPE_PARAM, instanceNetworkType);
+      return this;
+   }
+
+   public ListInstancesOptions privateIpAddresses(String... privateIpAddresses) {
+      String instanceIdsAsString = Joiner.on(",")
+              .join(Iterables.transform(Arrays.asList(privateIpAddresses), new Function<String, String>() {
+                 @Override
+                 public String apply(String s) {
+                    return new StringBuilder(s.length() + 1).append('"').append(s).append('"').toString();
+                 }
+              }));
+      queryParameters.put(PRIVATE_IP_ADDRESSES_PARAM, String.format("[%s]", instanceIdsAsString));
+      return this;
+   }
+
+   public ListInstancesOptions innerIpAddresses(String... innerIpAddresses) {
+      String instanceIdsAsString = Joiner.on(",")
+              .join(Iterables.transform(Arrays.asList(innerIpAddresses), new Function<String, String>() {
+                 @Override
+                 public String apply(String s) {
+                    return new StringBuilder(s.length() + 1).append('"').append(s).append('"').toString();
+                 }
+              }));
+      queryParameters.put(INNER_IP_ADDRESSES_PARAM, String.format("[%s]", instanceIdsAsString));
+      return this;
+   }
+
+   public ListInstancesOptions publicIpAddresses(String... publicIpAddresses) {
+      String instanceIdsAsString = Joiner.on(",")
+              .join(Iterables.transform(Arrays.asList(publicIpAddresses), new Function<String, String>() {
+                 @Override
+                 public String apply(String s) {
+                    return new StringBuilder(s.length() + 1).append('"').append(s).append('"').toString();
```

Comment:
Extract this to a `quote` function or similar you can reuse and properly unit test.